### DISCUSSION TITLE
3.16.6 on Synology with DSM 7

### DISCUSSION THREAD

**Discussion by dozer1dave** [tmknight/docker-nordvpn#76] [Q&A]
Trying to get this container to work with no luck.  It is interesting that it authenticates with Nord but never connects.  It dies with "Whoops! We couldn't connect you to 'us10179'.  Please try again.  If problem persists, contact our customer support."  Anyone else have this working on Synology 7?  I guess docker repository does not maintain older builds or I would check an older version.  Thanks

Dave

DSM 7.2-64570 Update 3
    Docker version 20.10.23, build 876964a
    iptables v1.8.3 (legacy)

Log below:

2023/10/13 10:59:16 | stdout | 2023-10-13 17:59:16+0000	INFO	Periodic VPN connection refresh is disabled
-- | -- | --
2023/10/13 10:59:16 | stdout | 2023-10-13 17:59:16+0000	INFO	Startup complete
2023/10/13 10:59:16 | stdout | 2023-10-13 17:59:16+0000	INFO	Enabling connection to IPv4 network 192.168.1.0/24
2023/10/13 10:59:16 | stdout | ########################
2023/10/13 10:59:16 | stdout | -     -    Firewall is already set to 'disabled'.
2023/10/13 10:59:16 | stdout | -     - \ \| /    Connecting to United States #10179 (us10179.nordvpn.com)
2023/10/13 10:59:13 | stdout | The recommended server is US10179
2023/10/13 10:59:13 | stdout | Login successful; connection now in progress...
2023/10/13 10:59:13 | stdout | -     -    Subnet 192.168.1.0/24 is allowlisted successfully.
2023/10/13 10:59:13 | stdout | -     -    Subnet 172.30.0.0/24 is allowlisted successfully.
2023/10/13 10:59:13 | stdout | -     -    Port 8000 (UDP\|TCP) is allowlisted successfully.
2023/10/13 10:59:13 | stdout | -     -    Port 8002 (UDP\|TCP) is allowlisted successfully.
2023/10/13 10:59:13 | stdout | -     -    Protocol is already set to UDP.
2023/10/13 10:59:13 | stdout | -     -    Technology is set to 'OPENVPN' successfully.
2023/10/13 10:59:12 | stdout | -     -    DNS is set to '103.86.96.100' successfully.
2023/10/13 10:59:12 | stdout | -    NordVPN Version 3.16.6
2023/10/13 10:59:12 | stdout | ########################
2023/10/13 10:59:12 | stdout | 2023-10-13 17:59:12+0000	INFO	Configuring and connecting NordVPN...
2023/10/13 10:59:12 | stdout | 2023-10-13 17:59:12+0000	INFO	Enabling connection to secure interface and docker network
2023/10/13 10:59:12 | stdout | 2023-10-13 17:59:12+0000	INFO	Firewall is up; traffic will be routed through vpn
2023/10/13 10:59:10 | stdout | 2023-10-13 17:59:10+0000	INFO	Waiting for nord daemon to start...

**Comment by tmknight** [comment]
The last entry suggests nordvpnd is not running. First step to troubleshoot is to docker run the container without any env with bash as the command. Step through manually starting the daemon, logging in, etc

**Reply by dozer1dave** [reply]
Here is the log from doing what you suggest.  Problem is during nord_connect.  Bolded part I think is most relevant.  I am not using ipv6 so not sure what is going on there...

2023/10/16 20:29:41 [Info] starting vpn                                          
Connecting to United States #10197 (us10197.nordvpn.com)                         
debug: 2023-10-16 20:29:41 DEPRECATED OPTION: --cipher set to 'AES-256-CBC' but missing in --data-ciphers (AES-256-GCM:AES-128-GCM). Future OpenVPN version will ignore —cipher for cipher negotiations. Add 'AES-256-CBC' to --data-ciphers or change --cipher 'AES-256-CBC' to --data-ciphers-fallback 'AES-256-CBC' to silence this warning.                                                                     
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 DEPRECATED OPTION: --cipher set to 'AES-256-CBC' but missing in --data-ciphers (AES-256-GCM:AES-128-GCM). Future OpenVPN version will ignore --cipher for cipher negotiations. Add 'AES-256-CBC' to --data-ciphers or change --cipher 'AES-256-CBC' to --data-ciphers-fallback 'AES-256-CBC' to silence this warning.                                                 
debug: 2023-10-16 20:29:41 OpenVPN 2.5.8 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Sep 18 2023                         
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 OpenVPN 2.5.8 x86_64-pc-linux-gnu [SSL (OpenSSL)] [LZO] [LZ4] [EPOLL] [MH/PKTINFO] [AEAD] built on Sep 18 2023     
debug: 2023-10-16 20:29:41 library versions: OpenSSL 1.1.1t  7 Feb 2023, LZO 2.10
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 library versions: OpenSSL 1.1.1t  7 Feb 2023, LZO 2.10                                                             
debug: 2023-10-16 20:29:41 MANAGEMENT: Connected to management server at /run/norddvpn/nordvpn-openvpn.sock                                                        
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 MANAGEMENT: Connected to management server at /run/nordvpn/nordvpn-openvpn.sock                                    
debug: 2023-10-16 20:29:41 MANAGEMENT: CMD 'state on'                            
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 MANAGEMENT: CMD 'state on'        
debug: 2023-10-16 20:29:41 MANAGEMENT: CMD 'username "Auth" ***************'                                                                          
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 MANAGEMENT: CMD 'username "Auth" ***************'                    debug: 2023-10-16 20:29:41 MANAGEMENT: CMD 'password [...]'  
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 MANAGEMENT: CMD 'password [...]'  
debug: 2023-10-16 20:29:41 Outgoing Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication                                   
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 Outgoing Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication               
debug: 2023-10-16 20:29:41 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication                                   
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 Incoming Control Channel Authentication: Using 512 bit message hash 'SHA512' for HMAC authentication               
debug: 2023-10-16 20:29:41 TCP/UDP: Preserving recently used remote address: [AF_INET]154.47.26.204:1194                                                          
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 TCP/UDP: Preserving recently used remote address: [AF_INET]154.47.26.204:1194                                      
debug: 2023-10-16 20:29:41 Socket Buffers: R=[212992->212992] S=[212992->212992] 
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 Socket Buffers: R=[212992->212992] S=[212992->212992]                                                              
debug: 2023-10-16 20:29:41 UDP link local: (not bound)                           
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 UDP link local: (not bound)       
debug: 2023-10-16 20:29:41 UDP link remote: [AF_INET]154.47.26.204:1194          
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 UDP link remote: [AF_INET]154.47.26.204:1194                                                                       
debug: 2023-10-16 20:29:41 MANAGEMENT: >STATE:1697488181,WAIT,,,,,,              
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 MANAGEMENT: >STATE:1697488181,WAIT,,,,,,                                                                           
debug: 2023-10-16 20:29:41 MANAGEMENT: >STATE:1697488181,AUTH,,,,,,              
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 MANAGEMENT: >STATE:1697488181,AUTH,,,,,,                                                                           
debug: 2023-10-16 20:29:41 TLS: Initial packet from [AF_INET]154.47.26.204:1194, sid=8c25811c 9638bfe0                                                            
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 TLS: Initial packet from [AF_INET]154.47.26.204:1194, sid=8c25811c 9638bfe0                                        
debug: 2023-10-16 20:29:41 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this                               
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 WARNING: this configuration may cache passwords in memory -- use the auth-nocache option to prevent this           
debug: 2023-10-16 20:29:41 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA                                                                                
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 VERIFY OK: depth=2, C=PA, O=NordVPN, CN=NordVPN Root CA                                                            
debug: 2023-10-16 20:29:41 VERIFY OK: depth=1, O=NordVPN, CN=NordVPN CA8         
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 VERIFY OK: depth=1, O=NordVPN, CN=NordVPN CA8                                                                      
debug: 2023-10-16 20:29:41 VERIFY KU OK                                          
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 VERIFY KU OK                      
debug: 2023-10-16 20:29:41 Validating certificate extended key usage             
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 Validating certificate extended key usage                                                                          
debug: 2023-10-16 20:29:41 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication                                      
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 ++ Certificate has EKU (str) TLS Web Server Authentication, expects TLS Web Server Authentication                  
debug: 2023-10-16 20:29:41 VERIFY EKU OK                                         
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 VERIFY EKU OK                     
debug: 2023-10-16 20:29:41 VERIFY X509NAME OK: CN=us10197.nordvpn.com            
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 VERIFY X509NAME OK: CN=us10197.nordvpn.com                                                                         
debug: 2023-10-16 20:29:41 VERIFY OK: depth=0, CN=us10197.nordvpn.com            
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 VERIFY OK: depth=0, CN=us10197.nordvpn.com                                                                         
debug: 2023-10-16 20:29:41 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1633', remote='link-mtu 1634'                                              
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 WARNING: 'link-mtu' is used inconsistently, local='link-mtu 1633', remote='link-mtu 1634'                          
debug: 2023-10-16 20:29:41 WARNING: 'comp-lzo' is present in remote config but missing in local config, remote='comp-lzo'                                         
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 WARNING: 'comp-lzo' is present in remote config but missing in local config, remote='comp-lzo'                     
debug: 2023-10-16 20:29:41 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, peer certificate: 4096 bit RSA, signature: RSA-SHA512                               
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 Control Channel: TLSv1.3, cipher TLSv1.3 TLS_AES_256_GCM_SHA384, peer certificate: 4096 bit RSA, signature: RSA-SHA512                                                                              
debug: 2023-10-16 20:29:41 [us10197.nordvpn.com] Peer Connection Initiated with [AF_INET]154.47.26.204:1194                                                       
2023/10/16 20:29:41 [INFO] 2023-10-16 20:29:41 [us10197.nordvpn.com] Peer Connection Initiated with [AF_INET]154.47.26.204:1194                                   
debug: 2023-10-16 20:29:42 MANAGEMENT: >STATE:1697488182,GET_CONFIG,,,,,,        
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 MANAGEMENT: >STATE:1697488182,GET_CONFIG,,,,,,                                                                     
debug: 2023-10-16 20:29:42 SENT CONTROL [us10197.nordvpn.com]: 'PUSH_REQUEST' (status=1)                                                                          
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 SENT CONTROL [us10197.nordvpn.com]: 'PUSH_REQUEST' (status=1)                                                      
debug: 2023-10-16 20:29:42 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,dhcp-option DNS 103.86.96.100,dhcp-option DNS 103.86.99.100,explicit-exit-notify,comp-lzo no,route-gateway 10.8.0.1,topology subnet,ping 60,ping-restart 180,ifconfig 10.8.0.126 255.255.0.0,peer-id 5,cipher AES-256-GCM'             
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 PUSH: Received control message: 'PUSH_REPLY,redirect-gateway def1,dhcp-option DNS 103.86.96.100,dhcp-option DNS 103.86.99.100,explicit-exit-notify,comp-lzo no,route-gateway 10.8.0.1,topology subnet,ping 60,ping-restart 180,ifconfig 10.8.0.126 255.255.0.0,peer-id 5,cipher AES-256-GCM'                                                                          
debug: 2023-10-16 20:29:42 Pushed option removed by filter: 'redirect-gateway def1'                                                                               
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 Pushed option removed by filter: 'redirect-gateway def1'                                                           
debug: 2023-10-16 20:29:42 OPTIONS IMPORT: timers and/or timeouts modified       
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 OPTIONS IMPORT: timers and/or timeouts modified                                                                    
debug: 2023-10-16 20:29:42 OPTIONS IMPORT: explicit notify parm(s) modified      
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 OPTIONS IMPORT: explicit notify parm(s) modified                                                                   
debug: 2023-10-16 20:29:42 OPTIONS IMPORT: compression parms modified            
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 OPTIONS IMPORT: compression parms modified                                                                         
debug: 2023-10-16 20:29:42 OPTIONS IMPORT: --ifconfig/up options modified        
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 OPTIONS IMPORT: --ifconfig/up options modified 
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 OPTIONS IMPORT: route-related options modified                                                                     
debug: 2023-10-16 20:29:42 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified                                                                       
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 OPTIONS IMPORT: --ip-win32 and/or --dhcp-option options modified                                                   
debug: 2023-10-16 20:29:42 OPTIONS IMPORT: peer-id set                           
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 OPTIONS IMPORT: peer-id set       
debug: 2023-10-16 20:29:42 OPTIONS IMPORT: adjusting link_mtu to 1656            
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 OPTIONS IMPORT: adjusting link_mtu to 1656                                                                         
debug: 2023-10-16 20:29:42 OPTIONS IMPORT: data channel crypto options modified  
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 OPTIONS IMPORT: data channel crypto options modified                                                               
debug: 2023-10-16 20:29:42 Data Channel: using negotiated cipher 'AES-256-GCM'   
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 Data Channel: using negotiated cipher 'AES-256-GCM'                                                                
debug: 2023-10-16 20:29:42 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key                                                               
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 Outgoing Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key                                           
debug: 2023-10-16 20:29:42 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key                                                               
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 Incoming Data Channel: Cipher 'AES-256-GCM' initialized with 256 bit key
debug: 2023-10-16 20:29:42 TUN/TAP device nordtun opened                         
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 TUN/TAP device nordtun opened     
debug: 2023-10-16 20:29:42 MANAGEMENT: >STATE:1697488182,ASSIGN_IP,,10.8.0.126,,,,                                                                                
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 MANAGEMENT: >STATE:1697488182,ASSIGN_IP,,10.8.0.126,,,,                                                            
debug: 2023-10-16 20:29:42 /sbin/ip link set dev nordtun up mtu 1500             
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 /sbin/ip link set dev nordtun up mtu 1500                                                                          
debug: 2023-10-16 20:29:42 /sbin/ip link set dev nordtun up                      
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 /sbin/ip link set dev nordtun up  
debug: 2023-10-16 20:29:42 /sbin/ip addr add dev nordtun 10.8.0.126/16           
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 /sbin/ip addr add dev nordtun 10.8.0.126/16                                                                        
debug: 2023-10-16 20:29:42 Initialization Sequence Completed                     
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 Initialization Sequence Completed 
debug: 2023-10-16 20:29:42 MANAGEMENT: >STATE:1697488182,CONNECTED,SUCCESS,10.8.0.126,154.47.26.204,1194,,                                                        
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 MANAGEMENT: >STATE:1697488182,CONNECTED,SUCCESS,10.8.0.126,154.47.26.204,1194,,                                    
2023/10/16 20:29:42 [Info] Setting the routing rules up                          
2023/10/16 20:29:42 [Info] starting network configuration                        
**2023/10/16 20:29:42 [Warning] executing 'ip -6 rule del not from all fwmark 57841' command: exit status 2: RTNETLINK answers: No such file or directory**                                        
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 MANAGEMENT: CMD 'pid'             
debug: 2023-10-16 20:29:42 MANAGEMENT: CMD 'signal "SIGINT"'                     
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 MANAGEMENT: CMD 'signal "SIGINT"' 
debug: 2023-10-16 20:29:42 SIGTERM received, sending exit notification to peer   
2023/10/16 20:29:42 [INFO] 2023-10-16 20:29:42 SIGTERM received, sending exit notification to peer                                                                
debug: 2023-10-16 20:29:43 Closing TUN/TAP interface                             
2023/10/16 20:29:43 [INFO] 2023-10-16 20:29:43 Closing TUN/TAP interface         
debug: 2023-10-16 20:29:43 /sbin/ip addr del dev nordtun 10.8.0.126/16           
2023/10/16 20:29:43 [INFO] 2023-10-16 20:29:43 /sbin/ip addr del dev nordtun 10.8.0.126/16                                                                        
debug: 2023-10-16 20:29:43 SIGTERM[soft,exit-with notification] received, processexiting                                                                         
2023/10/16 20:29:43 [INFO] 2023-10-16 20:29:43 SIGTERM[soft,exit-with-notification] received, process exiting                                                     
debug: 2023-10-16 20:29:43 MANAGEMENT: >STATE:1697488183,EXITING,exit-with-notification,,,,,                                                                      
2023/10/16 20:29:43 [INFO] 2023-10-16 20:29:43 MANAGEMENT: >STATE:1697488183,EXITING,exit-with-notification,,,,,                                                  
2023/10/16 20:29:43 [Error] enabling allowlist routing: enabling allowlisting for subnets: iptables inserting rule: exit status 2: iptables v1.8.7 (legacy): Couldn't load match `comment':No such file or directory                               
                                                                                 
Try `iptables -h' or 'iptables --help' for more information.

**Reply by dozer1dave** [reply]
Tried disabling ipv6 via sysctl.....problem still exists

**Reply by samueljrodriguez** [reply]
@dozer1dave Did you ever get this working? I'm able to login but can't connect. Also on synology through docker.
